rust-runtime/aws-smithy-runtime/src/client/orchestrator/auth.rs

```diff
@@ -214,50 +208,67 @@ pub(super) fn sign_request(
+// `resolve_identity` skips endpoint resolution, ensuring that `try_attempt` follows
+// the correct resolution order: auth scheme → identity → endpoint.
+// If the marker is absent, `try_attempt` continues using
+// the legacy, incorrect resolution order: endpoint → auth scheme → identity.
 #[doc(hidden)]
 #[derive(Clone, Debug)]
-pub struct AuthSchemeOptionResolverHandlesEndpointResolution;
+pub struct AuthSchemeAndEndpointOrchestrationV2;
 
-impl Storable for AuthSchemeOptionResolverHandlesEndpointResolution {
+impl Storable for AuthSchemeAndEndpointOrchestrationV2 {
     type Storer = StoreReplace<Self>;
 }
 
-// Checks whether an `endpoint` resolved by `SharedEndpointResolver` in `runtime_components`,
-// using `EndpointResolverParams` from `cfg`, contains `scheme_id` in its `authSchemes` property.
-//
-// Returns `Ok` if the condition is met, with the `Ok` variant also containing the actual `endpoint` used for verification.
+// Conditionally return an `endpoint` resolved by `SharedEndpointResolver` in `runtime_components`
+// whose `authSchemes` property matches the given `scheme_id`
 //
-// This function exists for backward compatibility. SDKs generated with the `AuthSchemeOptionResolverHandlesEndpointResolution`
-// do not, by default, require an endpoint outside of `SharedAuthSchemeOptionResolver`,
-// so the function short-circuits and returns `Ok(None)`.
-async fn endpoint_auth_scheme_matches_configured_scheme_id(
+// Return the `Ok` variant in the following cases:
+// - If `AuthSchemeAndEndpointOrchestrationV2` is present in the config bag, indicating that the runtime doesn't require the functionality of this function.
+//   In this case, the function short-circuits and returns `Ok(None)`, as no endpoint needs resolution. Instead, a custom `AuthSchemeOptionResolver`
+//   should have resolved it internally for auth scheme selection.
+// - If the runtime uses the legacy auth scheme and endpoint orchestration and resolves an endpoint that matches the given condition,
+//   the function returns `Ok(endpoint)`, which will then be applied to the request.
+async fn legacy_try_resolve_endpoint(
     runtime_components: &RuntimeComponents,
     cfg: &ConfigBag,
     scheme_id: AuthSchemeId,
 ) -> Result<Option<Endpoint>, AuthOrchestrationError> {
-    if cfg
-        .load::<AuthSchemeOptionResolverHandlesEndpointResolution>()
-        .is_some()
-    {
-        // Short-circuit if auth scheme option resolver internally handles endpoint resolution
+    if cfg.load::<AuthSchemeAndEndpointOrchestrationV2>().is_some() {
+        // The orchestrator uses the correct auth scheme and endpoint resolution order,
+        // and no endpoint needs to be resolved within this function.
         return Ok(None);
     }
 
     let params = cfg
         .load::<EndpointResolverParams>()
         .expect("endpoint resolver params must be set");
 
+    tracing::debug!(endpoint_params = ?params, "resolving endpoint");
```

Comment:
This should have a different message as it makes it seem like we're resolving the endpoint for normal execution flow but really its for legacy concerns. Perhaps something like `using legacy resolution order, resolving endpoint for auth` or something.